src/public/modules/forms/admin/components/response-components/response-attachment.client.component.js

```diff
@@ -8,23 +9,24 @@ angular.module('forms').component('responseAttachmentComponent', {
     encryptionKey: '<',
   },
   controllerAs: 'vm',
-  controller: [
-    'Submissions',
-    '$timeout',
-    responseAttachmentComponentController,
-  ],
+  controller: ['$q', '$timeout', responseAttachmentComponentController],
 })
 
-function responseAttachmentComponentController(Submissions, $timeout) {
+function responseAttachmentComponentController($q, $timeout) {
   const vm = this
 
   vm.downloadAndDecryptAttachment = function () {
     vm.hasDownloadError = false
-    Submissions.downloadAndDecryptAttachment(
-      vm.field.downloadUrl,
-      vm.encryptionKey.secretKey,
+    $q.when(
+      AdminSubmissionsService.downloadAndDecryptAttachment(
+        vm.field.downloadUrl,
+        vm.encryptionKey.secretKey,
+      ),
     )
       .then((bytesArray) => {
+        if (!bytesArray) {
+          vm.hasDownloadError = true
+        }
```

Comment:
a bit confused by this block now - the point was to avoid triggering the download right, so you would need to `return` early here? otherwise you might as well omit it and let the error fall through to the catch block

Reply:
thanks for the catch! fixed